oauthenticator/github.py

```diff
@@ -109,20 +155,41 @@ def authenticate(self, handler, data=None):
         resp = yield http_client.fetch(req)
         resp_json = json.loads(resp.body.decode('utf8', 'replace'))
 
-        username =  resp_json["login"]
+        username = resp_json["login"]
 
         # Check if user is a member of any whitelisted organizations.
         # This check is performed here, as it requires `access_token`.
         if self.github_organization_whitelist:
             for org in self.github_organization_whitelist:
                 user_in_org = yield self._check_organization_whitelist(org, username, access_token)
-                if user_in_org:
-                    return username
-            else:  # User not found in member list for any organisation
-                return None
-        else:  # no organization whitelisting
-            return username
-
+                if not user_in_org:
+                    # User not found in member list for any organisation
+                    return None
```

Comment:
We need to change the logic a little, obviously, since we need to fall through to set up auth_state, so....

```
        if self.github_organization_whitelist:
            for org in self.github_organization_whitelist:
                user_in_org = yield self._check_organization_whitelist(org, use\
rname, access_token)
                if user_in_org:
                    break
            else:  # User not found in member list for any organisation
                return None
```

Not a huge fan of `for/else` myself but this is what it's for.